src/mjolnir/shortcutbuilder.cc

```diff
@@ -69,6 +74,11 @@ bool EdgesMatch(const graph_tile_ptr& tile, const DirectedEdge* edge1, const Dir
   // NOTE: might want "better" bridge attribution. Seems most overpasses
   // get marked as a bridge and lead to less shortcuts - so we don't consider
   // bridge and tunnel here
+  // TODO(nils):
+  //   - links restriction could be removed? we don't care about it costing I think
+  //   - most other attributes (other than classification & surface) could also take the "most
+  //   restrictive"
+  //      for the entire shortcut
```

Comment:
actually: if the attribute is used to cost by **a factor** (i.e. it scales with length of the edge), then we do need to break up. but if it's like `destination_only` lump penalty we don't need to

Reply:
there can't be many (any?) destination_only motorways/trunks. I assume you are looking only at attributes that impact driving costing classes (pretty sure bicycle and pedestrian skip shortcuts).

Reply:
> there can't be many (any?) destination_only motorways/trunks

that's true. even tertiary would be rare I imagine (the least FRC we even contract)

Reply:
Was there a change to create shortcuts on secondary and tertiary? Or is my memory bad? I thought we only created shortcuts on level 0 (motorways, trunks, primary).

Reply:
Also level 1:)

Reply:
That’s why I think it’d make quite an impact